records/2023-11-24.system-status.md

```diff
@@ -0,0 +1,93 @@
+|     API     | DOWN     | The API is not responding                                                              |
+|             |          |                                                                                        |
+| File upload | UP       | The DD-API is responding to requests in 400ms or less                                  |
+| File upload | DEGRADED | The DD-API is responding to requests in more than 400ms                                |
+| File upload | DOWN     | The DD- API is not responding                                                          |
+|             |          |                                                                                        |
+|    Email    | UP       | Priority emails are sending within 60s, and Other emails are sending within 45m        |
+|    Email    | DEGRADED | Priority emails are taking longer than 60s, or Other emails are taking longer than 45m |
+|    Email    | DOWN     | No emails sent within the last 5 minutes                                               |
+|             |          |                                                                                        |
+|     SMS     | UP       | Priority SMSs are sending within 60s, and Other SMSs are sending within 45m            |
+|     SMS     | DEGRADED | Priority SMSs are taking longer than 60s, or Other SMSs are taking longer than 45m     |
+|     SMS     | DOWN     | No SMS messages sent within the last 5 minutes                                         |
+
+## Monitoring
+Each system will need to be monitored every 5m to determine which status it is in.
+
+## Systems architecture
+The website that shows system status needs to be deployed separately from the main GC Notify website so that we can continue showing status information even if the GC Notify website is down.  It should not live within our existing EKS node.
+
+### Front-end
+A new front-end website will be deployed to show the status information.  It will be served from a new domain, status.notification.canada.ca
+
+### Back-end
+A new back-end API wil be deployed that will be responsible for assessing the status of each component of the system.  
+
+#### Status checks
+- The website/API/File upload checks can all be done via simple HTTP requests to the respective systems.  No special permissions or rights are required. 
+    - **Question**: Can we leverage [the CDS status page](https://github.com/cds-snc/status-statut) for this?
+- For email and SMS, we will need to assess the actual sending time for each priority of each message type (email and SMS).  These checks will require read access to the notify database in order to calculate sending time.
+   - The healthcheck lambda should be extended to send a notification of each type and priority, and we can use these to assess the health of the email and sms sending functions.
+
+## Code architecture
```

Comment:
Another option would be to release a single admin pod to a standalone kube cluster. Though it would have to hit admin/api in the original eks cluster.